### ISSUE TITLE
error TS2307: Cannot find module

### ISSUE THREAD

**Issue by boardthatpowder** [pnpm/pnpm#1933]
### Code to reproduce the issue:

Using pnpm as the package manager for a typescript monorepo where we have a top level package.json containing global scripts and dependencies, with leaf package.json's per service.

Running `pnpm run build` in turn runs `tsc -b tsconfig.json` which throws the following error:

```
 > cat pnpm-debug.log
{
  "0 debug pnpm:scope": {
    "selected": 1,
    "workspacePrefix": "/Users/deanhart/git/cdf-ts/cdf-core"
  },
  "1 error pnpm": {
    "message": {
      "errno": 1,
      "code": "ELIFECYCLE",
      "pkgid": "cdf",
      "stage": "build",
      "script": "pnpm m run build",
      "pkgname": "cdf"
    },
    "err": {
      "name": "Error",
      "message": "cdf build: `pnpm m run build`\nExit status 1",
      "code": "ELIFECYCLE",
      "stack": "Error: cdf build: `pnpm m run build`\nExit status 1\n    at EventEmitter.proc.on (/Users/deanhart/.nvm/versions/node/v8.10.0/lib/node_modules/pnpm/lib/node_modules/@zkochan/npm-lifecycle/index.js:302:16)\n    at emitTwo (events.js:126:13)\n    at EventEmitter.emit (events.js:214:7)\n    at ChildProcess.<anonymous> (/Users/deanhart/.nvm/versions/node/v8.10.0/lib/node_modules/pnpm/lib/node_modules/@zkochan/npm-lifecycle/lib/spawn.js:55:14)\n    at emitTwo (events.js:126:13)\n    at ChildProcess.emit (events.js:214:7)\n    at maybeClose (internal/child_process.js:925:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)"
    }
  }
}
```

### Expected behavior:

Running `tsc -b tsconfig.json` as part of a package.json script successfully builds a project.

### Actual behavior:

Since `pnpm@3.5.7`, running `tsc` fails.  Reverting to `pnpm@3.5.3` works.

### Additional information:

 - `node -v` prints:  `v8.10.0`
 - Windows, OS X, or Linux?:  `OS X, and Linux`

**Comment by zkochan** [comment]
do you have a sample repo? We also use typescript but noticed no regressions

**Comment by boardthatpowder** [comment]
Pushed a sample [here]( https://github.com/boardthatpowder/pnpm-issue).

**Comment by zkochan** [comment]
I couldn't reproduce it. What it prints to the console in your case?

**Comment by Dionysusnu** [comment]
I've got same issue and steps to reproduce:

create a new folder
create file index.ts with contents
```ts
// getting the error in a bigger project for this module but it also errors here
import { Client } from 'discord.js';
// Add new Client() so it's not an unused import and actually gets included in the output
const client = new Client();
```
create file package.json with contents
```json
{
  "name": "pnpmtest",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
	  "discord.js": "11.5.1"
  },
  "devDependencies": {
	  "typescript": "3.4.3"
  }
}
```
run `pnpm install`
running it again says
```
Lockfile is up-to-date, resolution step is skipped
Already up-to-date
```
run `tsc index.ts --target esnext` --> error TS2307: Cannot find module 'discord.js'
Targets es3 and es5 do compile but later versions don't. Might be an issue on typescript's part.

**Comment by Dionysusnu** [comment]
Using the `--traceResolution` option I've found a difference in the module resolution.
ES3 outputs `Module resolution kind is not specified, using 'NodeJs'.`
ES6+ outputs `Module resolution kind is not specified, using 'Classic'.`
It seems like the Classic method does not look inside the folder `pnpmtest/node_modules/discord.js/`, only inside `pnpmtest/node_modules/@types/`

**Comment by Dionysusnu** [comment]
adding option `--moduleResolution node` fixes the error. I'm not sure if this is the same problem as the original by @boardthatpowder

**Comment by saurabhp-onit** [comment]
Where to put this option?

**Comment by Dionysusnu** [comment]
Either in tsconfig.json
or when running tsc
`tsc --moduleResolution node`

**Comment by zkochan** [comment]
All the typescript issues were fixed in pnpm v5.2
